File: src/bundler.go
package main

import (
  "bytes"
  "fmt"
  "os"
  "log"
  "os/exec"
  "strings"
)

func installNpmModules(name string, prefix string) {
  dir, errr := os.Getwd()
  if errr != nil {
    log.Fatal(errr)
    os.Exit(1)
  }
  cmdName := dir + "/node_modules/.bin/npm"
  cmdArgs := []string{"install", name, "--prefix",  prefix}
  cmd := exec.Command(cmdName, cmdArgs...)
  err := cmd.Run()
  if err != nil {
    log.Fatal(err)
  }
  cmd.Stdin = strings.NewReader("")
  var out bytes.Buffer
  cmd.Stdout = &out
  fmt.Printf("%q\n", out.String())
}

func main() {
  args := os.Args
  fmt.Printf(args[1], args[2])
  installNpmModules(args[1], args[2])
}

File: src/bundler.js

import request from 'request';
import fs from 'fs';
import rimraf from 'rimraf';
import appRoot from 'app-root-path';
import child_process from 'child_process';

export function checkIfBundleExists(res, path, verion, packageToCamelcase, callback) {
  fs.exists(path, exists => {
    if (!exists) {
      callback();
    }else {
      res.view('package',  {
        path: verion,
        name: packageToCamelcase,
        env: process.env.NODE_ENV
      });
    }
  })
}

export function requestNpmPackage(res, packageName, callback) {
  request(`https://registry.npmjs.org/${packageName}`, function (error, response, body) {
    if (!error && response.statusCode === 200) {
      return callback(JSON.parse(body));
    }else {
      return res(Boom.badImplementation(`npm packaging failed - ${error}`));
    }
  });
}

export function installNpmModules(res, name, config, callback) {
  return child_process.exec(`${appRoot.path}/node_modules/.bin/npm install ${name} --prefix ${config.prefix}`,
    error => {
      if (error !== null) {
        return res(Boom.badImplementation(`Server error: where was en error retrieving ${name} from npm`));
      }
      callback();
    });
}

export function buildWithWebpack(res, packageInstallPath, entry, packageToCamelcase, outPath, callback) {
  return child_process.exec(`${appRoot.path}/node_modules/.bin/webpack ${entry} --output-filename ${packageToCamelcase}.js -p  --context ${packageInstallPath} --output-library ${packageToCamelcase} --output-library-target var --output-path ${outPath}`,
    error => {
      if (error !== null) {
        return res(Boom.badImplementation(`Server error: where was en error retrieving ${name} from npm`));
      }
      callback();
    });
}

export function deleteModules(res, path, callback) {
  rimraf(path, error => {
    if (error) {
      return res(Boom.badImplementation(`Server error: deleting the modules`));
    }
    callback();
  })
}